bookwyrm/models/bookwyrm_export_job.py

```diff
@@ -0,0 +1,217 @@
+        # readthroughs
+        book_readthroughs = (
+            models.ReadThrough.objects.filter(user=user, book=book["id"])
+            .distinct()
+            .values()
+        )
+        book["readthroughs"] = list(book_readthroughs)
+        # shelves
+        shelf_books = models.ShelfBook.objects.filter(
+            user=user, book=book["id"]
+        ).distinct()
+        shelves_from_books = models.Shelf.objects.filter(
+            shelfbook__in=shelf_books, user=user
+        )
+
+        book["shelves"] = list(shelves_from_books.values())
+        book["shelf_books"] = {}
+
+        for shelf in shelves_from_books:
+            shelf_contents = models.ShelfBook.objects.filter(
+                user=user, shelf=shelf
+            ).distinct()
+
+            book["shelf_books"][shelf.identifier] = list(shelf_contents.values())
+
+        # book lists
+        book_lists = models.List.objects.filter(
+            books__in=[book["id"]], user=user
+        ).distinct()
+        book["lists"] = list(book_lists.values())
+        book["list_items"] = {}
+        for blist in book_lists:
+            list_items = models.ListItem.objects.filter(book_list=blist).distinct()
+            book["list_items"][blist.name] = list(list_items.values())
+
+        # reviews
+        reviews = models.Review.objects.filter(user=user, book=book["id"]).distinct()
+
+        book["reviews"] = list(reviews.values())
+
+        # comments
+        comments = models.Comment.objects.filter(user=user, book=book["id"]).distinct()
+
+        book["comments"] = list(comments.values())
+
+        # quotes
+        quotes = models.Quotation.objects.filter(user=user, book=book["id"]).distinct()
+
+        book["quotes"] = list(quotes.values())
+
+        # append everything
+        final_books.append(book)
+
+    # saved book lists
+    saved_lists = models.List.objects.filter(id__in=user.saved_lists.all()).distinct()
+    saved_lists = [l.remote_id for l in saved_lists]
+
+    # follows
+    follows = models.UserFollows.objects.filter(user_subject=user).distinct()
+    following = models.User.objects.filter(
+        userfollows_user_object__in=follows
+    ).distinct()
+    follows = [f.remote_id for f in following]
+
+    # blocks
+    blocks = models.UserBlocks.objects.filter(user_subject=user).distinct()
+    blocking = models.User.objects.filter(userblocks_user_object__in=blocks).distinct()
+
+    blocks = [b.remote_id for b in blocking]
+
+    data = {
+        "user": exported_user,
+        "goals": goals_list,
+        "books": final_books,
+        "saved_lists": saved_lists,
+        "follows": follows,
+        "blocked_users": blocks,
+    }
+
+    return DjangoJSONEncoder().encode(data)
+
+
+def get_books_for_user(user):
+    """Get all the books and editions related to a user
+    :returns: tuple of editions, books
+    """
+    all_books = models.Edition.viewer_aware_objects(user)
```

Comment:
Does this need to be viewer aware? I don't see the prefetched data being used

Reply:
Honestly I think I just copied this from somewhere else when I was initially seeing whether it might work. So this should be something like this instead:

`all_books = models.Edition.objects.filter(user=user)`?

Reply:
Yes, that looks right. In this case, `viewer_aware` preloads some user data about the books that makes the templates render more efficiently.